lein-cljfmt/src/leiningen/cljfmt.clj

```diff
@@ -59,20 +59,28 @@
      (apply check project (format-paths project))
      (main/abort "No project found and no source paths provided")))
   ([project path & paths]
-   (let [files   (mapcat (partial find-files project) (cons path paths))
-         flag    (atom 0)]
-     (doseq [f     files
-             :let  [original (slurp f)
-                    revised  (reformat-string project original)]
-             :when (not= original revised)]
-       (main/warn (project-path project f) "has incorrect formatting:")
-       (main/warn (format-diff project f original revised))
-       (swap! flag inc))
-     (if (zero? @flag)
+   (let [files             (mapcat (partial find-files project) (cons path paths))
+         incorrect-counter (atom 0)
+         failure-counter   (atom 0)]
```

Comment:
I don't think that `reduce` is appropriate unless this is reworked to return some sort of structure which returns counters rather than using local mutation. `loop` would be worse than `doseq` because at least `doseq` gives you the loop control structure for free.